### ISSUE TITLE
When will release mae pretrained weights?

### ISSUE THREAD

**Issue by shi-lt** [facebookresearch/hiera#14]

**Comment by UmaisZahid** [comment]
Any update on on the pretrained MAE release? Even just having the pretrained decoder weights would be quite useful.

**Comment by dbolya** [comment]
@UmaisZahid We are currently going back and forth about how to release these weights. May I ask, what do you plan to use the decoder for?

The issue is the models were trained with a normalized pixel loss (i.e., each patch that the decoder had to predict was normalized within that patch). Thus, using the pretrained decoder for reconstructing a masked image wouldn't look correct. Hence, we were planning on removing the decoder when releasing the weights (I believe MAE does the same thing).

Would the decoder still be useful in spite of that?

**Comment by UmaisZahid** [comment]
Thanks for the response @dbolya. 

I believe so yes. One use case for the decoder would be to further pretrain on another dataset. The ImageNet pretraining would act as a warm start or inductive bias of sorts in this case - which I imagine is particularly useful in the cases where the second pretraining dataset is smaller. 

You could release the decoder and perhaps remind people explicitly in the GitHub Readme what you just mentioned about the reconstructive loss - that would be very useful!

**Comment by dbolya** [comment]
Thanks for the context. We've decided to release all the pretrained weights with decoders and all!

The pretrained models are available in v0.1.2 (see the readme on how to use them). Closing the issue now, but feel free to reopen if you run into any trouble.

**Comment by abhishekaich27** [comment]
@dbolya Are these weights for MViT2 or modified MViT2 as proposed in the paper?

**Comment by dbolya** [comment]
@abhishekaich27 These weights are for Hiera, the final model that we end up with after making all modifications to MViTv2.

**Comment by abhishekaich27** [comment]
Thanks @dbolya! Do you plan to release or provide the MAE pre-trained MViT2 model?

**Comment by dbolya** [comment]
@abhishekaich27 No plans for that afaik.
